assembly-main/src/assembly/bin/che.sh

```diff
@@ -482,42 +493,57 @@ function call_catalina {
-    "${DOCKER}" exec che ${DOCKER_EXEC} &>/dev/null 2>&1 || DOCKER_EXIT=$? || true
+    echo -e "Stopping docker container named ${GREEN}${CONTAINER}${NC}...."
+    DOCKER_EXEC="sudo service docker stop && sudo //home/user/che/tomcat/bin/shutdown.sh"
+    "${DOCKER}" exec ${CONTAINER} ${DOCKER_EXEC} || DOCKER_EXIT=$? || true
 
-    echo -e "Stopping docker container named ${GREEN}che${NC}."
-    "${DOCKER}" stop che &>/dev/null 2>&1 || DOCKER_EXIT=$? || true
+    "${DOCKER}" stop --time=10 ${CONTAINER} &>/dev/null 2>&1 || DOCKER_EXIT=$? || true
   fi
+  return
 }
 
-function kill_and_launch_docker_che {
+kill_and_launch_docker_che () {
+
+  echo -e "A Docker container named ${GREEN}${CONTAINER}${NC} does not exist or duplicate conflict was discovered."
+  echo -e "Cleaning up any zombie containers named ${GREEN}${CONTAINER}${NC}..."
+
+  # Force remove any lingering zombie containers with the name ${CONTAINER}
+  "${DOCKER}" rm -f ${CONTAINER} &> /dev/null || true
 
-  echo -e "A Docker container for ${GREEN}che${NC} does not exist or duplicate conflict was discovered."
-  echo -e "Launching a new Docker container named ${GREEN}che${NC} from image ${GREEN}codenvy/che:${CHE_DOCKER_TAG}${NC} with Docker command:"
-  "${DOCKER}" kill che &> /dev/null || true
-  "${DOCKER}" rm che &> /dev/null || true
+  echo -e "Pulling the Docker image labeled ${GREEN}codenvy/che:${CHE_DOCKER_TAG}${NC}..."
+  "${DOCKER}" pull codenvy/che:${CHE_DOCKER_TAG} || true
+
+  echo -e "Launching a new Docker container named ${GREEN}${CONTAINER}${NC} from image ${GREEN}codenvy/che:${CHE_DOCKER_TAG}${NC}..."
 
   if ${WIN} || ${MAC} ; then
-     # This DOCKER_HOST is environment variable set by docker-machine - location of VM w/ Docker
-    set -x
-    "${DOCKER}" run --privileged -e DOCKER_MACHINE_HOST=${host} --name che -d -p ${CHE_PORT}:${CHE_PORT} --net=host codenvy/che:${CHE_DOCKER_TAG} bash -c 'true && sudo chown -R user:user /home/user/che && sudo service docker start && tail -f /dev/null' #&> /dev/null
+    DOCKER_PRINT_VALUE=${host}
   else
-    set -x
-    "${DOCKER}" run --privileged -e DOCKER_MACHINE_HOST=${DOCKER_MACHINE_HOST} --name che -d -p ${CHE_PORT}:${CHE_PORT} --net=host codenvy/che:${CHE_DOCKER_TAG} bash -c 'true && sudo chown -R user:user /home/user/che && sudo service docker start && tail -f /dev/null' #&> /dev/null
+    DOCKER_PRINT_VALUE=${DOCKER_MACHINE_HOST}
   fi
 
+  if ${USE_DEBUG}; then
+	  set -x
+  fi
+
+  # IDEX-4266 - Change launching of docker to avoid using dind
+  "${DOCKER}" run -v //var/run/docker.sock:/var/run/docker.sock \
+  -v //home/user/che/lib:/home/user/che/lib-copy \
+  -v //home/user/che/workspaces:/home/user/che/workspaces \
+  -v //home/user/che/tomcat/temp/local-storage:/home/user/che/tomcat/temp/local-storage \
+  -e DOCKER_MACHINE_HOST=${DOCKER_PRINT_VALUE} --name ${CONTAINER} -d -p ${CHE_PORT}:${CHE_PORT} --net=host codenvy/che:${CHE_DOCKER_TAG} \
+  bash -c "tail -f /dev/null" || DOCKER_EXIT=$? || true 
```

Comment:
`bash -c "tail -f /dev/null"` should be in Dockerfile CMD instruction. Why do we use it here?